hera_sim/adjustment.py

```diff
@@ -0,0 +1,974 @@
+    new_ant_2_array = np.empty(new_Nblts, dtype=int)
+    new_baseline_array = np.empty(new_Nblts, dtype=int)
+    new_uvw_array = np.empty((new_Nblts, 3), dtype=float)
+
+    # Rephase and update the data/metadata; some repeated code here.
+    for i, antpair in enumerate(target.get_antpairs()):
+        ant1, ant2 = antpair
+        this_slice = slice(i, None, target.Nbls)
+        old_blt = target._key2inds(antpair)[0][0]  # As a reference
+        this_uvw = target.uvw_array[old_blt]
+        this_baseline = target.baseline_array[old_blt]
+        this_integration_time = target.integration_time[old_blt]
+
+        # Update the metadata.
+        new_ant_1_array[this_slice] = ant1
+        new_ant_2_array[this_slice] = ant2
+        new_baseline_array[this_slice] = this_baseline
+        new_uvw_array[this_slice] = this_uvw
+        new_time_array[this_slice] = ref_times
+        new_integration_times[this_slice] = this_integration_time
+
+        # Update the data
+        for pol_ind, pol in enumerate(target.polarization_array):
+            pol = polnum2str(pol)
+            antpairpol = antpair + (pol,)
+            vis = data[antpairpol]
+            bl = bls[antpairpol]
+            new_data[this_slice, 0, :, pol_ind] = lst_rephase(
+                vis, bl, data.freqs, dlst, lat=lat, array=True
+            )
+
+    # Convert from HERAData object to UVData object
+    _uvd = UVData()
+    for attr in _uvd:
+        setattr(_uvd, attr, getattr(target, attr))
+    target = _uvd
+    del _uvd
+
+    # Times/LSTs need special treatment.
+    for ref_time, target_time in ref_to_target_time_map.items():
+        if target_time is None:
+            continue
+        this_slice = np.argwhere(new_time_array == ref_time).flatten()
+        new_lst_array[this_slice] = ref_time_to_lst_map[ref_time]
+
+    # Now update all of the data/metadata.
+    target.Ntimes = np.unique(new_time_array).size
+    target.Nblts = new_Nblts
+    target.time_array = new_time_array
+    target.lst_array = new_lst_array
+    target.integration_time = new_integration_times
+    target.ant_1_array = new_ant_1_array
+    target.ant_2_array = new_ant_2_array
+    target.baseline_array = new_baseline_array
+    target.uvw_array = new_uvw_array
+    target.data_array = new_data
+    target.flag_array = np.zeros(new_data.shape, dtype=bool)
+    target.nsample_array = np.ones(new_data.shape, dtype=float)
+    target.blt_order = None
+
+    # Convert to Simulator if needed, then return the result
+    if target_is_simulator:
+        target = Simulator(data=target)
+
+    return target
+
+
+def _validate_file_list(file_list, name="file list"):
+    """Ensure all entries in the file list are path-like objects."""
+    if not all(isinstance(item, (str, pathlib.Path)) for item in file_list):
+        raise TypeError(
+            f"{name} must be either a collection of path-like objects or a "
+            "UVData object or Simulator object"
```

Comment:
That may seem like the case since this function is only ever called in the context of `_to_uvdata` (barring one case, which I think I would like to change to use `_to_uvdata` instead of the routine it uses now), but the idea is that this function is only ever run when we already know the input object is not a UVData or Simulator object. In fact, if I enforce that this is only ever called in the context of `_to_uvdata`, then I could optionally just remove this message, since `_to_uvdata` gives its own error messages based on what kind of error it encounters.

Reply:
Yeah, I'd try to keep things localised -- if you're going to use `_to_uvdata`-specific information in this function, define it in `_to_uvdata`.

Reply:
OK, I've updated the error message here.